File: articles/type-systems-and-the-discipline-of-computational-representation/java/src/main/java/org/contentcatalyst/algorithms/TypeRepresentationAudit.java
package org.contentcatalyst.algorithms;

public class TypeRepresentationAudit {
  enum PublicationStatus { DRAFT, REVIEW, PUBLISHED, ARCHIVED }
  record ArticleRecord(String title, String slug, PublicationStatus status) {}

  public static void main(String[] args) {
    ArticleRecord record = new ArticleRecord(
      "Type Systems and the Discipline of Computational Representation",
      "type-systems-and-the-discipline-of-computational-representation",
      PublicationStatus.PUBLISHED
    );
    System.out.println("title,slug,status");
    System.out.println(record.title() + "," + record.slug() + "," + record.status());
  }
}

File: articles/type-systems-and-the-discipline-of-computational-representation/typescript/src/typeRepresentationAudit.ts
export type PublicationStatus = "draft" | "review" | "published" | "archived";
export type ArticleRecord = {
  title: string;
  slug: string;
  series: "Algorithms & Computational Reasoning";
  status: PublicationStatus;
  repositoryUrl: string;
};

export const article: ArticleRecord = {
  title: "Type Systems and the Discipline of Computational Representation",
  slug: "type-systems-and-the-discipline-of-computational-representation",
  series: "Algorithms & Computational Reasoning",
  status: "published",
  repositoryUrl: "https://github.com/Content-Catalyst-LLC/algorithms-computational-reasoning-code/tree/main/articles/type-systems-and-the-discipline-of-computational-representation/"
};
